### PR TITLE
fix: connect multiple accounts even if current account is connected

### PR DESCRIPTION
**Pull request by eswarasai**
[FRO-242](https://linear.app/fuel-network/issue/FRO-242/when-click-to-connect-show-popup-to-connect-accounts)

### CODE REVIEW

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/app/src/systems/DApp/machines/connectRequestMachine.ts [hunk 1]
```diff
@@ -183,26 +193,39 @@ export const connectRequestMachine = createMachine(
...
             },
           });
         },
       }),
     },
     guards: {
-      connectionConnected: (_, ev) => {
-        return !!((ev.data?.accounts.length || 0) >= 1);
+      connectionConnected: (ctx, ev) => {
+        return (ev.data?.accounts.length || 0) === ctx.totalAccounts;
```
Comment (matt-user) [inline packages/app/src/systems/DApp/machines/connectRequestMachine.ts:228]: @luizstacio should we also add a check here that `ctrx.totalAccounts >= 1`?

File: packages/app/src/systems/DApp/machines/connectRequestMachine.ts [hunk 2]
```diff
@@ -183,26 +193,39 @@ export const connectRequestMachine = createMachine(
...
+            });
           }
           return ConnectionService.addConnection({
             data: {
               origin: input.origin,
-              accounts: input.accounts,
-              favIconUrl: input.favIconUrl,
-              title: input.title,
+              accounts: input.selectedAddresses,
+              favIconUrl: input.favIconUrl!,
```
Comment (matt-user) [inline packages/app/src/systems/DApp/machines/connectRequestMachine.ts:219]: @luizstacio we should find some way to remove the `!` right?  The easiest way is to add a check such as
```
if (!input.favIconUrl || !input.title) throw Error ("...");
```
is there a better way

Comment (luizstacio) [reply packages/app/src/systems/DApp/machines/connectRequestMachine.ts:219]: In some cases, it's ok to use it, but I think in this case is not required, it should accept undefined...

File: packages/app/src/systems/DApp/machines/connectRequestMachine.ts (hunk 1 above)
Comment (luizstacio) [reply packages/app/src/systems/DApp/machines/connectRequestMachine.ts:228]: yeah 
```suggestion
        return (ev.data?.accounts.length || 0) === ctx.totalAccounts && ctx.totalAccounts >= 1
```

**Review by luizstacio — APPROVED** [review]
LGTM

**Pull request merged**